### DISCUSSION TITLE
Service Load Balancer

### DISCUSSION THREAD

**Discussion by zhoub** [k3s-io/k3s#4368] [Q&A]
Hi,

We have a service which consumes a lot of memory for data processing, which is written in GRPC.

It was exposed as Service with type `LoadBalancer` with external IP, I'd ask one question for this if possible, assume we have Node A, B, C, each of them has same Pod, and when client calls service at Node B with its IP, would service balancer forwards the request to another Node A or C, or just enqueue this request to Node B as requested ? What happened when Type is NodePort, is there anything different ?

Thank you very much !

**Comment by brandond** [comment]
The service doesn't do anything, it just forwards packets via iptables. If your grpc client is Kubernetes aware (can discover all the possible endpoints for the service via DNS) and supports multiple endpoints it might be able to do that, but Kubernetes itself will not.

**Reply by zhoub** [reply]
Hi,

Understood ! Many thanks for your reply !
